sw/device/lib/testing/rand_testutils.h

```diff
@@ -7,22 +7,97 @@
 
 #include <stdint.h>
 
+#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
+
+/**
+ * A random number generator testutil type.
+ *
+ * Provides ability to control and maintain a random number generator. A random
+ * number is produced via a mix of hardware-based (using on-device entropy
+ * source) and software-based (a simple PRNG) approaches. The initial seed
+ * value is fetched from the on-device entropy source. Following that, the PRNG
+ * kicks in and supplies random values for testing purposes. After a number of
+ * PRNG cycles, the LFSR can be freshly reseeded from the on-device entropy
+ * source. If this frequency is set to 0, then random data is always fetched
+ * from the entropy source.
```

Comment:
is it from the entropy source or from the ibex entropy registers? it'll be good to be explicit here because for better or worse we have an actual module called `entropy_src` heh.

Reply:
Hmm, I could have read `entropy_src.ENTROPY_DATA` directly also rather than take the CSRNG -> EDN -> rv_core_ibex route and fetch from `rv_core_ibex.RND_DATA` instead. Is there a preference for one over the other?

Reply:
o no, i think where you're reading is the preferred approach. No arguments from me.
I only meant that we should not use the words "entropy source" and specify this is from the ibex EDN endpoint to avoid any confusion.  

The implementation looks good to me.